### DISCUSSION TITLE
Deepsleep and telemetry

### DISCUSSION THREAD

**Discussion by EmilSodergren** [arendst/Tasmota#13976] [Q&A] [answered]
Hi!
I have a wemos d1 mini with an SHT30 shield that sends data via MQTT. I have set the DeepSleepTime=300 and the TelePeriod=10. 
This according to the Tasmota documentation about [DeepSleep](https://tasmota.github.io/docs/DeepSleep/#deepsleep-algorithm-general-timing). 
> If you want to minimize the time that the device is in operation, decrease TelePeriod down to 10 seconds.

The behaviour I get is that the wemos wakes from deep sleep and sends the temperature, waits for what seems to be ~20 seconds, sends another temperature and then goes back to sleep. 

The result of this is that I am getting a sawtooth curve with the first reading showing one temperature and the other showing that temperature plus 0.3°C after about 15-20 seconds. I suspect this is how much the wemos heats up when powered on.

![2021-dec- 9-Screenshot_1](https://user-images.githubusercontent.com/5655742/145455117-15b5c1ad-f7af-40e8-a605-d308b3a618e0.png)
![2021-dec- 9-Screenshot](https://user-images.githubusercontent.com/5655742/145455127-912f6997-cd91-427e-9e2b-402af0782f70.png)

Should it really send two readings every cycle?? There must be something I am not understanding correctly here, or?

**Comment by barbudor** [comment]
Logs?

**Comment by EmilSodergren** [comment]
00:00:00.001 HDW: ESP8285
00:00:00.050 CFG: Loaded from flash at FA, Count 106
00:00:00.056 CFG: CR 389/699, Busy 0
00:00:00.061 ROT: Mode 1
00:00:00.063 SRC: Restart
00:00:00.065 Project tasmota - Tasmota 10 Version 10.0.0(sensors)-2_7_4_9(2021-10-19T08:23:50)
00:00:00.078 SHT: Sensor did not ACK command
00:00:00.142 I2C: SHT3X found at 0x45
00:00:00.431 WIF: Checking connection...
00:00:00.432 WIF: Attempting connection...
00:00:00.769 WIF: Connecting to AP1 .....
00:00:01.752 WIF: Checking connection...
00:00:01.754 WIF: Connected
00:00:02.005 HTP: Web server active on outside-4851 with IP address 192.168.60.10
00:00:02.223 NTP: Synch time...
00:00:02.337 RTC: UTC 2021-12-09T20:39:46, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
21:39:46.000 NTP: Synched
21:39:47.044 MQT: Attempting connection...
21:39:47.061 MQT: Connected
21:39:47.064 MQT: tele/home/outside/LWT = Online (retained)
21:39:47.067 MQT: cmnd/home/outside/POWER =
21:39:47.068 MQT: Subscribe to cmnd/home/outside/#
21:39:47.072 MQT: Subscribe to cmnd/home/tasmotas/#
21:39:47.073 MQT: Subscribe to cmnd/DVES_F272F3_fb/#
21:39:47.079 MQT: Unsubscribe from homeassistant/status
21:39:47.144 SRC: MQTT
21:39:47.146 CMD: Grp 0, Cmnd 'DEEPSLEEPTIME', Idx 1, Len 3, Data '300'
21:39:47.151 MQT: stat/home/outside/RESULT = {"DeepSleepTime":300}
21:39:47.245 DSL: Reset wrong settings wakeup: 0, slip 10000
21:39:47.248 MQT: stat/home/outside/RESULT = {"DeepSleep":{"Time":"2021-12-09T21:40:00","Epoch":1639082400}}
00:00:00.001 HDW: ESP8285
00:00:00.050 CFG: Loaded from flash at FA, Count 106
00:00:00.056 CFG: CR 389/699, Busy 0
00:00:00.061 ROT: Mode 1
00:00:00.063 SRC: Restart
00:00:00.065 Project tasmota - Tasmota 10 Version 10.0.0(sensors)-2_7_4_9(2021-10-19T08:23:50)
00:00:00.078 SHT: Sensor did not ACK command
00:00:00.142 I2C: SHT3X found at 0x45
00:00:00.431 WIF: Checking connection...
00:00:00.432 WIF: Attempting connection...
00:00:00.768 WIF: Connecting to AP1 ...
00:00:01.801 WIF: Checking connection...
00:00:01.802 WIF: Connected
00:00:02.004 HTP: Web server active on outside-4851 with IP address 192.168.60.10
00:00:02.222 NTP: Synch time...
00:00:04.233 NTP: No reply
00:00:04.241 NTP: Synch time...
00:00:04.347 RTC: UTC 2021-12-09T20:40:08, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
21:40:08.000 NTP: Synched
21:40:09.002 MQT: Attempting connection...
21:40:09.025 MQT: Connected
21:40:09.028 MQT: tele/home/outside/LWT = Online (retained)
21:40:09.030 MQT: cmnd/home/outside/POWER =
21:40:09.032 MQT: Subscribe to cmnd/home/outside/#
21:40:09.033 MQT: Subscribe to cmnd/home/tasmotas/#
21:40:09.038 MQT: Subscribe to cmnd/DVES_F272F3_fb/#
21:40:09.042 MQT: Unsubscribe from homeassistant/status
21:40:09.252 SRC: MQTT
21:40:09.254 CMD: Grp 0, Cmnd 'DEEPSLEEPTIME', Idx 1, Len 3, Data '300'
21:40:09.259 MQT: stat/home/outside/RESULT = {"DeepSleepTime":300}
21:40:10.220 DSL: Reset wrong settings wakeup: 0, slip 10000
21:40:10.225 MQT: stat/home/outside/RESULT = {"DeepSleep":{"Time":"2021-12-09T21:45:00","Epoch":1639082700}}

**Reply by sfromis** [reply]
That bit of logs does not include the sensor payloads which seems to be your basic question.

Looks like you have a retained message or automation with a `DeepSleepTime 300` incoming right after MQTT connection. That may not be useful.....

**Reply by barbudor** [reply]
Indeed we don't see here what you claim

Please double check your Teleperiod. Normally with values 10 or 300, you should only have a single telemetry message and Tasmota should go back to sleep right after.

**Reply by EmilSodergren** [reply]
Oh, i thought you retained DeepSleep to make Tasmota read it and go into another sleep cycle... I try to remove it

**Comment by EmilSodergren** [comment, marked as answer]
I have double checked and the telemetry period is indeed 10, it does not seem to log anywhere though...

**Reply by barbudor** [reply]
Thiss is your current roblem:
```
07:55:07.239 SRC: MQTT
07:55:07.241 CMD: Grp 0, Cmnd 'DEEPSLEEPTIME', Idx 1, Len 3, Data '300'
07:55:07.246 MQT: stat/home/outside/RESULT = {"DeepSleepTime":300}
```

At every restart, Tasmota gets a new `DeepSleepTime 300` command which force Tasmota to recompute the next wake up time and immediately goes to sleep which going on the normal process.
Either you have this command as a retained message (as pointed by sfromis) or you have an automation which sends it everytime the device connects to MQTT.
You have to fix that first.

**Reply by EmilSodergren** [reply]
Ah yes... I did misunderstand how it should work. I thought retaining the DeepSleepTime was how you should do. When removing that it works as expected.

Thank you!
